src/casl/casl-ability.factory.ts

```diff
@@ -54,40 +54,199 @@ export type AppAbility = MongoAbility<PossibleAbilities, Conditions>;
 
 @Injectable()
 export class CaslAbilityFactory {
-  createForUser(user: JWTUser) {
+  accessEndpointForUser(user: JWTUser, subject: string) {
+    switch (subject) {
+      case "datasets":
+        return this.accessDatasetEndpointForUser(user);
+      case "elastic-search":
+        return this.accessElasticSearchEndpointForUser(user);
+      case "jobs":
```

Comment:
I think the naming can be simplified as `endpointAccess`, `datasetEndpointAccess` etc, which to me is clear enough. 
What do you think?

Reply:
yes, that might be better :)